pandas/tests/window/test_rolling.py

```diff
@@ -1781,3 +1781,21 @@ def test_step_not_integer_raises():
 def test_step_not_positive_raises():
     with pytest.raises(ValueError, match="step must be >= 0"):
         DataFrame(range(2)).rolling(1, step=-1)
+
+
+def test_rolling_center_axis_1():
+    df = DataFrame(
+        {"a": [1, 1, 0, 0, 0, 1], "b": [1, 0, 0, 1, 0, 0], "c": [1, 0, 0, 1, 0, 1]}
+    )
+
+    result = df.rolling(window=3, axis=1, win_type="boxcar", center=True).sum()
+
+    expected = DataFrame(
+        {
+            "a": [None, None, None, None, None, None],
+            "b": [3, 1, 0, 2, 0, 2],
+            "c": [None, None, None, None, None, None],
+        }
+    )
+
+    tm.assert_frame_equal(result, expected, check_dtype=False)
```

Comment:
Could you build the expected DataFrame with the correct dtype so you don't have to specify `check_dtype=False`?

Reply:
Just Did :)